src/esp/sensor/OrthoCamera.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <Magnum/ImageView.h>
+#include <Magnum/Math/Algorithms/GramSchmidt.h>
+#include <Magnum/PixelFormat.h>
+
+#include "OrthoCamera.h"
+#include "esp/gfx/DepthUnprojection.h"
+#include "esp/gfx/Renderer.h"
+#include "esp/sim/Simulator.h"
+
+namespace esp {
+namespace sensor {
+
+OrthoCamera::OrthoCamera(scene::SceneNode& orthoCameraNode,
+                         const sensor::SensorSpec::ptr& spec)
+    : sensor::VisualSensor(orthoCameraNode, spec) {
+  setProjectionParameters(spec);
+}
+
+void OrthoCamera::setProjectionParameters(const SensorSpec::ptr& spec) {
+  ASSERT(spec != nullptr);
+  width_ = spec_->resolution[1];
+  height_ = spec_->resolution[0];
+  near_ = std::atof(spec_->parameters.at("near").c_str());
+  far_ = std::atof(spec_->parameters.at("far").c_str());
+  scale_ = std::atof(spec_->parameters.at("ortho_scale").c_str());
+}
+
+OrthoCamera& OrthoCamera::setProjectionMatrix(gfx::RenderCamera& targetCamera) {
+  // use ortho version of function
+  targetCamera.setOrthoProjectionMatrix(width_, height_, near_, far_, scale_);
+  return *this;
+}
+
+OrthoCamera& OrthoCamera::setTransformationMatrix(
```

Comment:
My biggest concern is that there is a lot of code duplication (from PinholeCamera) in this file.
We may need to figure out a way to solve this problem.

If you lift them into the visualSensor class, be careful. For each function, we need to think twice. Would all visual sensors need this function?

@mosra : Thoughts?

Reply:
Yes, i was concerned about the duplication as well, but I did not want to promote anything to visualSensor if they were not appropriate for all visual sensors.  I also knew the redesign is pending and didn't want to mess with that.

Reply:
Not a problem at all. In my opinion, this PR is part of the "camera redesign" project, not a blocker. :)
I think the only difference between the `PinholeCamera` and the `OrthoCamera` is the projection, whether it is perspective or orthogonal. Correct? 
IF it is correct, my idea is that we rename `PinholeCamera` to `CameraSensor` for example, which has the `enum ProjectionType`  properly set to `ProjectionType::Perspective` or `ProjectionType::Orthogonal`. 

Did I miss anything that cannot be merged?

Reply:
I like @bigbike 's suggestion, `CameraSensor` with different projection types.

It's a little messy how to handle `hfov_` and `scale_` members which only apply to one or the other projection type, but that hardly justifies making two separate classes.

Reply:
Should these really be abstracted out in the class sense? Or should the code be placed in a different file and then use std::bind or lambdas instead.

Reply:
I don't like the duplication either, this makes it a hell to maintain.

> It's a little messy how to handle `hfov_` and `scale_` members which only apply to one or the other projection type, but that hardly justifies making two separate classes.

You can express the perspective camera [in terms of near projection plane size](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1Matrix4.html#aeed8f8ff2daedc28b7e3638e8e4ded47), which I'd say is pretty much equivalent to the orthographic projection size.